mvpa2/tests/test_rsa.py

```diff
@@ -165,4 +165,55 @@ def test_PDistTargetSimilaritySearchlight():
     # just because we are here and we can
     # Actually here for some reason assert_array_lequal gave me a trouble
     assert_true(np.all(sl_both.samples[1] <= 1.0))
-    assert_true(np.all(0 <= sl_both.samples[1]))
\ No newline at end of file
+    assert_true(np.all(0 <= sl_both.samples[1]))
+
+
+def test_Regression():
+    # a very correlated dataset
+    corrdata = np.array([[1, 2], [10, 20], [-1, -2], [-10, -20]])
+    # a perfect predictor
+    perfect_pred = np.array([0, 2, 2, 2, 2, 0])
+
+    ds = Dataset(corrdata)
+
+    reg_types = ['lasso', 'ridge']
+
+    # assert it pukes because predictor is not of the right shape
+    assert_raises(ValueError, Regression, perfect_pred)
+
+    # now make it right
+    perfect_pred = np.atleast_2d(perfect_pred).T
+    for reg_type in reg_types:
+        regr = Regression(perfect_pred, alpha=0, fit_intercept=False,
+                          rank_data=False, normalize=False, method=reg_type)
+    coefs = regr(ds)
+    assert_almost_equal(coefs.samples, 1.)
+
+    # what if we select some items?
+    keep_pairss = [range(3), [1], np.arange(3)]
+    for reg_type in reg_types:
+        for keep_pairs in keep_pairss:
+            regr = Regression(perfect_pred, keep_pairs=keep_pairs, alpha=0,
+                              fit_intercept=False, rank_data=False, normalize=False,
+                              method=reg_type)
+            coefs = regr(ds)
+            assert_almost_equal(coefs.samples, 1.)
+
+    # make a less perfect predictor
+    bad_pred = np.ones((6, 1))
+    predictors = np.hstack((perfect_pred, bad_pred))
+
+    # check it works with combination of parameters
+    for fit_intercept, rank_data, normalize, reg_type in \
+            zip([True, False], [True, False], [True, False], reg_types):
+        regr = Regression(predictors, alpha=1,
+                               fit_intercept=fit_intercept, rank_data=rank_data,
+                               normalize=normalize, method=reg_type)
+        coefs = regr(ds)
+        # check we get all the coefficients we need
+        wanted_samples = 3 if fit_intercept else 2
+        assert_equal(coefs.nsamples, wanted_samples)
```

Comment:
Could you add a more stringent test please? Right now it would seem that the tests would pass if regr(ds) returns a vector with all ones irrespective of the input. Preferably some hard-coded values that detect a chance in output of the function, or a minimal implementation to compute the expected output values. Less preferably at least some smoke test.